### PR TITLE
Stateful failover example

### PR DESCRIPTION
**Pull request by vanyarock01**

### CODE REVIEW

**Review by vasiliy-t — CHANGES_REQUESTED** [review]
I suggest to only reference relevant doc pages. And documenting that there is no direct support from operator and this is manual.

**Review by vasiliy-t — APPROVED** [review]

File: doc/cartridge_kubernetes_guide/index.rst [hunk 1]
```diff
@@ -699,6 +699,83 @@ If you need to remove the Tarantool Kubernetes operator, execute:
...
       persistentvolumeclaim "www-storages-0-0" deleted
 
+
+.. _cartridge_kubernetes_cluster_management_failover:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Failover
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Failover - replicaset leader selection mechanism. You can read more about it `here <https://www.tarantool.io/en/doc/latest/book/cartridge/topics/failover/>`_.
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:709]: There's no such document. https://www.tarantool.io/en/doc/latest/book/cartridge/topics/failover

If you want to make an internal link, use the [`:doc:` or `:ref:` link syntax](https://www.tarantool.io/en/doc/latest/contributing/docs/markup/#linking-to-document).

File: doc/cartridge_kubernetes_guide/index.rst [hunk 2]
```diff
@@ -699,6 +699,83 @@ If you need to remove the Tarantool Kubernetes operator, execute:
...
+      CPUallocation: 0.1
+      MemtxMemoryMB: 256
+      RolesToAssign:
+        - vshard-storage
+        - metrics
+        - failover-coordinator # added role
+
+.. NOTE::
+
+   Ability to update the roles is available in the Tarantool operator version > 0.0.8
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:746]: > version 0.0.8 or later.

File: doc/cartridge_kubernetes_guide/index.rst [hunk 3]
```diff
@@ -699,6 +699,83 @@ If you need to remove the Tarantool Kubernetes operator, execute:
...
+Eventual mode
+********************************************************************************
+
+Default mode. Uses SWIM protocol to detect failures.
+
+********************************************************************************
+Stateful mode
+********************************************************************************
+
+Uses external storage for coordination. To work, you need to enable a ``failover-coordinator`` role on several instances.
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:725]: It's a good practice to break lines by sentences. This way the source is better readable and diffs are more informative.

File: doc/cartridge_kubernetes_guide/index.rst [hunk 4]
```diff
@@ -699,6 +699,83 @@ If you need to remove the Tarantool Kubernetes operator, execute:
...
+
+To do this, add the role in *values.yml* to the description of the replicasets:
+
+.. code-block:: yaml
+
+  RoleConfig:
```
Comment (NickVolynkin) [inline doc/cartridge_kubernetes_guide/index.rst:731]: Not enough indentation. Both the `..  code-block` directive and the code example [should be indented by 4 spaces](https://www.tarantool.io/en/doc/latest/contributing/docs/markup/#wrapping-text-and-indentation). Here code example starts on 2 spaces, directive on 3 — this just won't build.

**Review by NickVolynkin — APPROVED** [review]

File: doc/cartridge_kubernetes_guide/index.rst (hunk 1 above)
Comment (vanyarock01) [reply doc/cartridge_kubernetes_guide/index.rst:709]: I want to save working links in github guide version.

**Pull request merged**